src/repositories/anchor-repository.ts

```diff
@@ -0,0 +1,39 @@
+import { EntityRepository, InsertResult } from "typeorm";
+import { BaseRepository } from "typeorm-transactional-cls-hooked";
+
+import Context from "../context";
+import Contextual from "../contextual";
+import { Anchor } from "../models/anchor";
+import { Request } from "../models/request";
+
+@EntityRepository(Anchor)
+export default class AnchorRepository extends BaseRepository<Anchor> implements Contextual {
+
+  setContext(context: Context): void {}
```

Comment:
It is expected this is a no-op? Seems strange to me to have a class implement an interface with no apparent use for it?

Reply:
The idea is that only `Contextual` instances will be available in the `Context` for dependency lookup.

Reply:
I think this could be helped by a more descriptive comment for the Contextual interface that clarifies that instances are global singletons managed by the Context object.